src/openrct2/ride/Vehicle.cpp

```diff
@@ -9321,17 +9331,20 @@ void Vehicle::UpdateCrossings() const
     uint8_t freeCount = travellingForwards && status != Vehicle::Status::Departing ? 3 : 1;
     while (freeCount-- > 0)
     {
-        auto* pathElement = MapGetPathElementAt(TileCoordsXYZ(CoordsXYZ{ xyElement, xyElement.element->GetBaseZ() }));
-        if (pathElement != nullptr)
+        if (travellingForwards)
```

Comment:
I know this is just moved, not introduced here, but it made me wonder. Is `travellingForwards` set for vehicles in shuttle mode?

Reply:
Only for the train head. This is why the `if` statement at line 9231 exacts (see the comment above that).

Reply:
Alright! That sounds pretty hacky, actually, but cleaning that up is way out of scope for this PR.